src/main/kotlin/com/example/delivery/model/MongoOrder.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.delivery.model
+
+import com.example.delivery.dto.response.OrderDTO
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.TypeAlias
+import org.springframework.data.mongodb.core.mapping.Document
+
+@TypeAlias("Order")
+@Document(collection = MongoOrder.COLLECTION_NAME)
+data class MongoOrder(
+    @Id val id: String? = null,
+    val items: Map<String, Int>?,
+    val totalPrice: Double?,
+    val shipmentDetails: ShipmentDetails?,
+    val status: Status? = Status.NEW) {
+    enum class Status {
+        NEW, SHIPMENT, COMPLETED, CANCELED, UNKNOWN
+    }
+    companion object {
+        const val COLLECTION_NAME = "order"
+    }
+
+    fun toDTO(): OrderDTO = OrderDTO(
```

Comment:
IMHO, You overloading this Mongo class by redundant logic. Make mappers instead of trying map inside data class. Same is true where you mapping to Mongo in your DTO